apps/studio/src/components/Datatable/Datatable.tsx

```diff
@@ -0,0 +1,167 @@
+          <Flex
+            bottom={0}
+            flex={1}
+            left={0}
+            pos="fixed"
+            right={0}
+            top={0}
+            w="100vw"
+            zIndex={2}
+          >
+            <Box m="auto">
+              <Spinner />
+            </Box>
+          </Flex>
+        </>
+      )}
+      <Box overflow={overflow} sx={styles.container}>
+        <Table sx={{ tableLayout: "fixed" }} {...tableProps} pos="relative">
+          <Thead>
+            {instance.getHeaderGroups().map((headerGroup) => (
+              <Tr
+                key={headerGroup.id}
+                // To toggle _groupHover styles to show divider when header is hovered.
+                data-group
+                borderBottomWidth="1px"
+              >
+                {headerGroup.headers.map((header) => (
+                  <Th
+                    key={header.id}
+                    pos="relative"
+                    px={0}
+                    style={{
+                      width:
+                        header.getSize() === Number.MAX_SAFE_INTEGER
+                          ? "auto"
+                          : header.getSize() === Number.MIN_SAFE_INTEGER
+                            ? "fit-content"
+                            : header.getSize(),
+                    }}
+                  >
+                    <Flex align="center">
+                      {flexRender(
+                        header.column.columnDef.header,
+                        header.getContext(),
+                      )}
+                    </Flex>
+                  </Th>
+                ))}
+              </Tr>
+            ))}
+          </Thead>
+          <Tbody>
+            {rows.length === 0 && emptyPlaceholder}
+            {rows.map((row) => {
+              return (
+                <Tr key={row.id} borderBottomWidth="1px">
+                  {row.getVisibleCells().map((cell) => {
+                    return (
+                      <Td
+                        key={cell.id}
+                        verticalAlign="center"
+                        {...tablePropOverrides?.[cell.column.id]?.td}
+                        {...tablePropOverrides?.[cell.id]?.td}
+                      >
+                        {flexRender(
+                          cell.column.columnDef.cell,
+                          cell.getContext(),
+                        )}
+                      </Td>
+                    )
+                  })}
+                </Tr>
+              )
+            })}
+          </Tbody>
+        </Table>
+      </Box>
+      <Flex py="1rem" gap="1rem">
+        {totalRowCountString && (
+          <Text textStyle="caption-2" color="base.content.medium">
+            {totalRowCountString}
+          </Text>
+        )}
+        {pagination && (
+          <Flex ml="auto">
+            <DatatablePagination
+              instance={instance}
+              totalRowCount={totalRowCount}
+            />
+          </Flex>
+        )}
```

Comment:
these 2 props (`totalRowCountString` and `totalRowCount`) seem to only be used here and the abstraction of `Datatable` became abit bloated because it now has to handle pagination in component as well. 

what do you feel about just cutting pagination out of datatable and letting consumers choose this behaviour?

Reply:
we not building a design system, so no need for components to be so generic lol. We can always refactor this when we get multiple distinct usecases of datatable (though i suspect we should be consistent with our tables)